### PR TITLE
Fix encoding global variables in imported modules

### LINKED ISSUE #38759: [Bug]: Runtime Error Throws for Generated Client Operation
### Description

I have tried to solve the Ballerina Hackathon problem_3_5. 

I have used the OpenAPI tool to generate the clients and then call the API. When I tried to call the `sales` API, it throws the following error:

```shell
java.lang.ClassFormatError: Illegal field name "$Client_$get$._quarter" in class balHack/problem_3_5/1/main

    at java.lang.ClassLoader:defineClass1(ClassLoader.java:-2)
    java.lang.ClassLoader:defineClass(ClassLoader.java:1017)
    java.security.SecureClassLoader:defineClass(SecureClassLoader.java:174)
    java.net.URLClassLoader:defineClass(URLClassLoader.java:555)
    java.net.URLClassLoader$1:run(URLClassLoader.java:458)
    java.net.URLClassLoader$1:run(URLClassLoader.java:452)
    java.security.AccessController:doPrivileged(AccessController.java:-2)
    java.net.URLClassLoader:findClass(URLClassLoader.java:451)
    java.lang.ClassLoader:loadClass(ClassLoader.java:589)
    java.lang.ClassLoader:loadClass(ClassLoader.java:522)
    balHack.problem_3_5$test.1.tests.tests:testFindTopCustomers(tests-tests:10)
    jdk.internal.reflect.NativeMethodAccessorImpl:invoke0(NativeMethodAccessorImpl.java:-2)
    jdk.internal.reflect.NativeMethodAccessorImpl:invoke(NativeMethodAccessorImpl.java:62)
    jdk.internal.reflect.DelegatingMethodAccessorImpl:invoke(DelegatingMethodAccessorImpl.java:43)
    java.lang.reflect.Method:invoke(Method.java:566)
    org.ballerinalang.test.runtime.entity.TesterinaFunction:lambda$runOnSchedule$0(TesterinaFunction.java:109)
    io.ballerina.runtime.internal.scheduling.SchedulerItem:execute(SchedulerItem.java:54)
    io.ballerina.runtime.internal.scheduling.Scheduler:run(Scheduler.java:324)
    io.ballerina.runtime.internal.scheduling.Scheduler:runSafely(Scheduler.java:291)
    java.lang.Thread:run(Thread.java:829)
```

Tried the same code with Ballerina 2201.2.3, it works fine.

### Steps to Reproduce

Generate the clients in the Ballerina Hackathon problem_3_5.

Then use the following code:

```ballerina
import problem_3_5.customers;
import problem_3_5.sales;

function findTopXCustomers(Quarter[] quarters, int x) returns customers:Customer[]|error {
    final sales:Client salesClient = check new ("http://localhost:8080/sales");
    sales:Sales[] sales = check salesClient->/(x);
    return [];
}
```

Check the same code with Ballerina 2201.2.3, it works without this error.

### Affected Version(s)

Ballerina 2201.3.0 (Swan Lake Update 3) RC2
RC1 is also affected by this.

### OS, DB, other environment details and versions

_No response_

### Related area

-> Runtime

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by Nadeeshan96**
$title
Fixes #38759

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java [hunk 1]
```diff
@@ -427,29 +427,20 @@ public void generateVarLoad(MethodVisitor mv, BIRNode.BIRVariableDcl varDcl, int
...
                 String varName = varDcl.name.value;
                 PackageID moduleId = ((BIRNode.BIRGlobalVariableDcl) varDcl).pkgId;
                 String pkgName = JvmCodeGenUtil.getPackageName(moduleId);
                 String className = jvmPackageGen.lookupGlobalVarClassName(pkgName, varName);
                 String typeSig = getTypeDesc(bType);
                 mv.visitFieldInsn(GETSTATIC, className, varName, typeSig);
                 return;
-            }
+            default:
+                break;
```
Comment (gabilang) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java:443]: Is the `default` case with `break` needed here?

Comment (Nadeeshan96) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java:443]: Yes, it is considered as a good coding practice to always add the default case in switch case. Sonarlint suggests that.

Comment (gabilang) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java:443]: We are not practicing that in most of the places in our code base (when default case does not seem necessary). We can adopt this in future as a coding best practice.

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**